src/Flights.groovy

```diff
@@ -1,106 +1,161 @@
         println("Passengers to Sofia - ${passengersToSof}")
 
-        def cap = flights.every {(double)it.passengers / (double)it.capacity >= 0.5}
-        println("Is every flight full at least 50% of its capacity? ${cap ? "Yes" : "No"} ")
+        boolean isAnyFlightUnderHalfCapacity = flights.every {(double)it.passengers / (double)it.capacity >= 0.5}
+        println("Is every flight full at least 50% of its capacity? ${isAnyFlightUnderHalfCapacity ? "Yes" : "No"} ")
 
         // Is there a flight to London that is running late
-        def londonLate = flights.findAll {it.destination == "LON" }.any { it.runningLate}
-        println("Is there a flight to London running late? ${londonLate ? "Yes" : "No"}")
+        boolean isAnyFlightToLondonLate = flights.findAll {it.destination == airports['London'] }.any { it.runningLate}
+        println("Is there a flight to London running late? ${isAnyFlightToLondonLate ? "Yes" : "No"}")
+        println("Flights to London running late")
+        flights.findAll {it.runningLate && it.destination == airports['London'] }
+            .each {println(it.toString())}
+
 
 
-        def pas = flights.findAll { it.runningLate == true}.collect { it.passengers }.sum()
-        println("Passengers running late ${pas}")
+        int passengersLate = flights.findAll { it.runningLate == true}.collect { it.passengers }.sum()
+        println("Passengers running late ${passengersLate}")
 
         println("All flights to Sofia that are running late:")
-        flights.findAll { it.destination == "SOF" && it.runningLate == true}.each { it.description() }
+        flights.findAll { it.destination == airports['Sofia'] && it.runningLate == true}.each { println(it)}
 
         println("The first flight that is running late and more than 95% of the capacity is full")
-        flights.findAll { it.runningLate == true }
-                .find { (double)it.passengers / (double)it.capacity > 0.95 }
-                .each { it.description() }
+        println(flights.findAll { it.runningLate == true }
+                .find { (double)it.passengers / (double)it.capacity > 0.95 }.toString())
 
         println("All flights from Sofia")
-        flights.findAll { it.startingLocation == "SOF"}.each { it.description() }
+        flights.findAll { it.from == airports['Sofia']}.each { println(it.toString()) }
 
         println("Sorted by day of arrival ascending")
-        flights.sort {a, b -> a.arrivalAt.getDayOfMonth() - b.arrivalAt.getDayOfMonth()}.each {it.description()}
+        flights.sort { it.arrival }
+                .each { println(it.toString()) }
 
         println("Sorted by day of arrival descending")
-        flights.sort {a, b -> b.arrivalAt.getDayOfMonth() - a.arrivalAt.getDayOfMonth()}.each {it.description()}
+        flights.sort {a, b -> b.arrival.isEqual(a.arrival) ? 0 : b.arrival.isBefore(a.arrival) ? -1 : 1}
```

Comment:
always remember that there is Comparable and .compareTo() method when sorting. Makes life easier